rocketpy/AeroSurface.py

```diff
@@ -368,11 +342,96 @@ def evaluate_geometrical_parameters(self):
+
+        # Calculate a function to find the tangential intersection point between the circle and nosecone curve.
+        def find_x_intercept(x):
+            return x + self.y_nosecone(x) * self.y_nosecone.differentiate(x)
+
+        # Calculate a function to find the radius of the nosecone curve
+        def find_radius(x):
+            return (self.y_nosecone(x) ** 2 + (x - find_x_intercept(x)) ** 2) ** 0.5
+
+        # Check bluffness circle and choose whether to use it or not
+        if self.bluffness is None or self.bluffness == 0:
+            # Set up parameters to continue without bluffness
+            r_circle, circle_center, x_init = 0, 0, 0
+        else:
+            # Calculate circle radius
+            r_circle = self.bluffness * self.base_radius
+            if self.kind == "elliptical":
+                # Calculate a function to set up a circle at the starting position to test bluffness
+                def test_circle(x):
+                    return np.sqrt(r_circle**2 - (x - r_circle) ** 2)
+
+                # Check if bluffness circle is too small
+                if test_circle(1e-03) <= self.y_nosecone(1e-03):
+                    # Raise a warning
+                    warnings.warn(
+                        "WARNING: The chosen bluffness ratio is too small for "
+                        "the selected nose cone category, thereby the effective "
+                        "bluffness will be 0."
+                    )
+                    # Set up parameters to continue without bluffness
+                    r_circle, circle_center, x_init = 0, 0, 0
+                else:
+                    # Find the intersection point between circle and nosecone curve
+                    x_init = fsolve(lambda x: find_radius(x) - r_circle, r_circle)[0]
+                    circle_center = find_x_intercept(x_init)
+            else:
+                # Find the intersection point between circle and nosecone curve
+                x_init = fsolve(lambda x: find_radius(x) - r_circle, r_circle)[0]
+                circle_center = find_x_intercept(x_init)
+
+        # Calculate a function to create the circle at the correct position
+        def create_circle(x):
+            return abs(r_circle**2 - (x - circle_center) ** 2) ** 0.5
+
+        # Define a function for the final shape of the curve with a circle at the tip
+        def final_shape(x):
+            return self.y_nosecone(x) if x >= x_init else create_circle(x)
+
+        # Vectorize the final_shape function
+        final_shape_vec = np.vectorize(final_shape)
+
+        # Create the vectors X and Y with the points of the curve
+        nosecone_x = (self.length - (circle_center - r_circle)) * (
+            np.linspace(0, 1, n) ** p
+        )
+        nosecone_y = final_shape_vec(nosecone_x + (circle_center - r_circle))
+
+        # Evaluate final geometry parameters
+        self.shape_vec = (nosecone_x, nosecone_y)
```

Comment:
tuples are immutable in python, but np.array are mutable.

Here we are defining a tuple of 2 np.arrays.

I believe it could be better to define as a list (or an array) instead of a tuple, 

One may also suggest to go ahead and define it as a rocketpy.Function object, but this should not work so well in the case of TrapezoidalFins, so I'd recommed not using Function for now.

Reply:
Are we ever going to mutate the shape_vec?

I'll change into a list just for safety

Also, I think we could use `PiecewiseFunction` for the fins drawings too. The problem is that making the plots with Matplotlib requires the vectors to be in a specific format, which is different from the format accepted in the Function class.

Reply:
> Are we ever going to mutate the shape_vec?

I'm not sure, but I think we could need to do some operations with this vector when trying to draw the entire rocket, but let's see.

> I'll change into a list just for safety
> 
> Also, I think we could use `PiecewiseFunction` for the fins drawings too. The problem is that making the plots with Matplotlib requires the vectors to be in a specific format, which is different from the format accepted in the Function class.

Alright, this is something to add to the #222 discussion